---
layout: default
pagination:
  enabled: true
  collection: posts
  permalink: /page/:num/
  per_page: 3
  sort_field: date
  sort_reverse: true
  trail:
    before: 1 # The number of links before the current page
    after: 3  # The number of links after the current page
---

<!DOCTYPE html>
<html>

  <head>
  <meta charset="UTF-8">
  <meta http-equiv="content-language" content="en">
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <meta name="viewport" content="width=device-width initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">

  <title>CS 502</title>
  <meta name="description" content="CS 502 - Deep Reinforcement Learning and Control - Carnegie Mellon University - Spring 2021
">

  <link rel="shortcut icon" href="https://upload.wikimedia.org/wikipedia/commons/thumb/9/96/University_of_Illinois_at_Chicago_circle_logo.svg/1200px-University_of_Illinois_at_Chicago_circle_logo.svg.png">

  <link rel="stylesheet" href="https://cmudeeprl.github.io/403_website/assets/css/main.css">
  <link rel="canonical" href="/403_website/">


  <!-- Load Latex JS -->
  <script src="https://cdn.jsdelivr.net/npm/latex.js@0.11.1/dist/latex.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/latex.js@0.11.1/dist/latex.component.js"></script>

</head>


  <body>

    <!-- <header class="site-header"> -->

  <div class="wrapper">

    <!--
    <span class="site-title">
      <a class="page-link" href="https://cmudeeprl.github.io/403_website/">10-403 Deep RL</a>
    </span>

    <nav class="site-nav">
      <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
              <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
              <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

      <div class="trigger">
        <a class="page-link" href="https://cmudeeprl.github.io/403_website">Home</a>
        <a class="page-link" href="/403_website/logistics/">Logistics</a>
        <a class="page-link" href="/403_website/lectures/">Lectures</a>
        <a class="page-link" href="/403_website/calendar/">Calendar</a>
        <a class="page-link" href="/403_website/homework/">Homework</a>
      </div>
    </nav>-->

  </div>


    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <header class="post-header about">
    <img class="logo" src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/96/University_of_Illinois_at_Chicago_circle_logo.svg/1200px-University_of_Illinois_at_Chicago_circle_logo.svg.png", height="120">
    <h1 class="post-title"><b>A</b>lgorithms in <b>C</b>omputational <b>M</b>olecular <b>B</b>iology</h1>
    <h2 class="post-description clearfix">CS 502 • Spring 2025 • University of Illinois Chicago</h2>
  </header>


  <article class="post-content <b>D</b>eep <b>R</b>einforcement <b>L</b>earning clearfix">
    <p><strong>Introduction</strong></p>
    <p>The revolutionary development of genomics has led to an immense volume of digital genetic information, offering vast opportunities for biological and medical discoveries. To analyze this data, advanced computational methods have become essential, giving rise to the field of Computational Molecular Biology. This field attracts computer scientists because genes, proteins, and genomes can be viewed as strings of symbols, making techniques from text processing applicable. Moreover, large-scale projects like the Human Genome Project and ENCODE have generated massive datasets, posing complex computational challenges. Despite heuristic solutions, many optimization problems remain intractable, with limited proofs of their accuracy.
</p>
<p>
The course will be focused on the design and analysis of efficient (combinatorial) algorithms for important problems in computational molecular biology. The format of the course will include lectures by the instructor, class discussion, directed reading, and student presentations or projects. The exact format will depend on the size of enrolment and student background. We emphasize mathematics, algorithms, and data structures instead of biological implications and applications, although some relevant biological background and motivations will be discussed. We may also have some guest speakers to talk about their research problems.
</p>

<hr />

<p><strong>Logistics</strong></p>
<ul>
<p>
<li><strong>Lectures:</strong> Tuesday and Thursday 2:00-3:15 pm in Lecture Center A6</li>
<li><strong>HW submission:</strong> <a href="https://www.gradescope.com" target="\_blank">Gradescope</a>
<!-- - **Contact:** For external enquiries, personal matters or in emergencies, you can email Brynn at
bedmunds@andrew.cmu.edu. --></li>
</p>
</ul>

<p><strong>Textbooks</strong></p>

<p>No textbook is required, but the following textbooks may be helpful as references.</p>
<ul>
  <li>Bioinformatics Algorithms: An Active Learning Approach<br>
Phillip Compeau and Pavel Pevzner<br>
Active Learning Publishers, 2018<br><br>
</li>
  <li>An Introduction to Bioinformatics Algorithms<br>
Neil C. Jones and Pavel Pevzner<br>
MIT Press, 2004<br><br>
</li>
  <li>Introduction to Computational Biology: Maps, Sequences and Genomes<br>
Michael S. Waterman<br>
Chapman Hall, 1995<br>
</li>
</ul>

<p><strong>Prerequisites</strong></p>
<ul>
  <li>Prerequisites: CS 401 Computer Algorithms I</li>
  <li>Recommended background: CS 501 and some exposure to basic biology.</li>
</ul>

<p><strong>Syllabus</strong></p>
<ul>
  <li>Introductory lecture (1 lecture)</li>
  <li>DNA mapping (1.5 lecture)</li>
  <li>Motif finding (2 lectures)</li>
  <li>Genome rearrangements (2 lectures)</li>
  <li>Gene prediction (1 lecture)</li>
  <li>HMM (1.5 lecture)</li>
  <li>Sequence alignment (2.5 lectures)</li>
  <li>Multiple sequence alignment (1.5 lectures)</li>
  <li>Graph algorithms for DNA sequencing (1 lecture)</li>
  <li>Reconstruction of evolutionary trees (1 or 2 lectures)</li>
  <li>DL for protein structure prediction (2 lectures)</li>
  <li>DL for single cell genomics (2 lectures)</li>
</li>
</ul>

<hr />

<p><strong>Grading</strong></p>
<p>Grading will be based upon three problem sets containing either programming questions or written questions (50%), a presentation (40%), and participation (10%). Attendance in lecture is important as the class moves quickly and you will need to be present.
</p>

<p><strong>Presentation</strong></p>
<p>Choose a paper from the Proceedings of RECOMB 2023 or 2024. Reserve your spot early using the provided sign-up sheet. Presentations will begin on April 1st and continue through the last week of instruction, ending on Thursday, May 1st. Please email your PowerPoint file to me at least two days before your scheduled presentation. Each presentation should last approximately 20 minutes, followed by 5 minutes for audience questions.
</p>
<p>
Ensure your presentation covers the following sections: (1) Biological background (~3 minutes): introduce the biological context. For example, if the paper discusses protein structure prediction, explain what proteins are, the significance of their structure, how it is determined, and why it matters. (2) Computational problem overview (~2 minutes): highlight the main computational problems addressed in the paper and the prediction target. (3) Existing approaches (~2 minutes): briefly discuss existing algorithms or methods used to solve the stated problem. (4) Proposed method (~5 minutes): describe the method introduced in the paper in detail. (5) Results (~5 minutes): summarize the key findings and results presented in the paper. (6) Future directions (~3 minutes): conclude with potential future research directions based on the paper's work.
</p>

<hr />
<!-- Staff  -->
<div class="instructors clearfix">

<div class="instructor-profile-two-col">
  <a href="https://www.cs.cmu.edu/~katef/" target="_blank"><img src="/assets/img/prof_pic_1.jpg" /></a>
  <ul class="instructor-info">
    <li><span>Instructor</span> <a href="https://www.cs.cmu.edu/~katef/" target="_blank">Hao Chen</a></li>
    <li><span>Email:</span> chenhao [AT] uic [DOT] edu</li>
    <li><span>Office hours:</span> Thursday 12:30 - 1:30 pm</li>
  </ul>
</div>
</div>
  </article>



  <!-- Announcements  -->
  <!-- <div class="news">
  <h2>Announcements</h2>

    <table>


    </table>

</div>
 -->

</div>



    <!-- Load jQuery -->
<script src="//code.jquery.com/jquery-1.12.4.min.js"></script>

<!-- Load Common JS -->
<script src="/assets/js/common.js"></script>


<!-- Load KaTeX -->
<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.1/katex.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/KaTeX/0.10.1/katex.min.js"></script>
<script src="https://github.com/cmudeeprl/403_website/tree/master/assets/js/katex.js"></script>



<!-- Load Anchor JS -->
<script src="//cdnjs.cloudflare.com/ajax/libs/anchor-js/3.2.2/anchor.min.js"></script>
<script>
  anchors.options.visible = 'hover';
  anchors.add('article h2, article h3, article h4, article h5, article h6');
</script>



<!-- Adjust LaTeX JS -->
<script src="https://github.com/cmudeeprl/403_website/tree/master/assets/js/latex.js"></script>


<!-- Include custom icon fonts -->
<link rel="stylesheet" href="https://github.com/cmudeeprl/403_website/tree/master/assets/css/fontawesome-all.min.css">
<link rel="stylesheet" href="https://github.com/cmudeeprl/403_website/tree/master/assets/css/academicons.min.css">

<!-- Google Analytics -->
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','//www.google-analytics.com/analytics.js','ga');

ga('create', '', 'auto');
ga('send', 'pageview');
</script>


  </body>

</html>
